searchcore/src/vespa/searchcore/proton/persistenceengine/document_iterator.cpp

```diff
@@ -91,18 +100,24 @@ DocumentIterator::DocumentIterator(const storage::spi::Bucket &bucket,
 
 DocumentIterator::~DocumentIterator() = default;
 
+void
+DocumentIterator::add(const IPersistenceHandler *handler, IDocumentRetriever::SP retriever)
```

Comment:
Consider passing the document type name directly, to avoid adding a coupling between DocumentIterator and IPersistenceHandler. Otherwise LGTM.

Reply:
Done